src/compiler/checker.ts

```diff
@@ -90,6 +90,9 @@ module ts {
         let unknownType = createIntrinsicType(TypeFlags.Any, "unknown");
 
         let emptyObjectType = createAnonymousType(undefined, emptySymbols, emptyArray, emptyArray, undefined, undefined);
+        let emptyGenericType = <GenericType><ObjectType>createAnonymousType(undefined, emptySymbols, emptyArray, emptyArray, undefined, undefined);

```

Comment:
I think you could just use `emptyObjectType` and add an `instantiations` property to it.